assets/app/scripts/directives/fromFile.js

```diff
@@ -236,9 +236,15 @@ angular.module("openshiftConsole")
           return DataService.get(resourceGroupVersion, item.metadata.name, $scope.context, {errorNotification: false}).then(
             // resource does exist
             function(resource) {
-              if (item.kind !== "Template") {
+              // For Template update it's Objects, Paramaters and Labels fields. For other resources update their Spec.
+              if (item.kind === "Template") {
+                resource.objects = item.objects;
+                resource.parameters = item.parameters;
+                resource.labels = item.labels;
```

Comment:
@spadgett `oc replace` will replace all three mentioned fields (wont update/add, just replace the whole field).
I was hesitating to do this as well, but couldn't think about better solution.